sound/soc/sof/sof-audio.c

```diff
@@ -43,12 +43,36 @@ static int sof_kcontrol_setup(struct snd_sof_dev *sdev, struct snd_sof_control *
 	return ret;
 }
 
-static int sof_dai_config_setup(struct snd_sof_dev *sdev, struct snd_sof_dai *dai)
+static struct sof_ipc_dai_config* sof_dai_config_match(struct snd_sof_dev *sdev,
+						       struct snd_sof_dai *dai,
+						       struct sof_ipc_pcm_params *ipc_pcm)
 {
+	int sample_rate = ipc_pcm->params.rate;
+	int num_conf = dai->num_conf;
+	int i;
+
+	/* match for ssp sample rate and fsync for now, otherwise use default config */
+	for (i = 0; i < num_conf; i++) {
```

Comment:
how about to check sample bit depth also ?  for example using 16 bit pipeline may save much memory than 24bit.

Reply:
yes, this was just very simple experiment. Main goal was to get the multiple configs parsed and saved to the driver. So yes we should think carefully how the matching could be done more generic, could we perhaps configure the matching also.. So I was doing this for testing bt a2dp vs. sco connection -> we just change the ssp params if you open the pcm with 48kHz or 8kHz.

Reply:
DAI config should be set by a enum kcontrol. This way UCM can set it. The first enum ID can always be ```automatic``` in that this function will pick setting based on rate etc.